File: arrays_strings/remove_duplicates/go/remove_duplicates.go
package remove_duplicates

func RemoveDuplicates(nums []int) int {
	l := len(nums)
	for j := 0; j < l; j++ {
		for k := j+1; k < l; k++ {
			if nums[k] == nums[j] {
				_ = append(nums[:k], nums[k+1:]...)
				l--
				k--
			}
		}
	}

	return l
}

File: arrays_strings/remove_duplicates/python/remove_duplicates.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        nums_map = {x: i for i, x in enumerate(nums)}
        for i, x in enumerate(nums_map.keys()):
            nums[i] = x

        return len(nums[:len(nums_map)])
